extensions/wikidata/src/org/openrefine/wikidata/commands/PreviewWikibaseSchemaCommand.java

```diff
@@ -0,0 +1,132 @@
+ * FITNESS FOR A PARTICULAR PURPOSE AND NONINFRINGEMENT. IN NO EVENT SHALL THE
+ * AUTHORS OR COPYRIGHT HOLDERS BE LIABLE FOR ANY CLAIM, DAMAGES OR OTHER
+ * LIABILITY, WHETHER IN AN ACTION OF CONTRACT, TORT OR OTHERWISE, ARISING FROM,
+ * OUT OF OR IN CONNECTION WITH THE SOFTWARE OR THE USE OR OTHER DEALINGS IN THE
+ * SOFTWARE.
+ ******************************************************************************/
+
+package org.openrefine.wikidata.commands;
+
+import java.io.IOException;
+import java.io.StringWriter;
+import java.util.List;
+import java.util.Properties;
+
+import javax.servlet.ServletException;
+import javax.servlet.http.HttpServletRequest;
+import javax.servlet.http.HttpServletResponse;
+
+import org.json.JSONException;
+import org.json.JSONObject;
+import org.json.JSONWriter;
+
+import com.google.refine.browsing.Engine;
+import com.google.refine.commands.Command;
+
+import org.openrefine.wikidata.exporters.QuickStatementsExporter;
+import org.openrefine.wikidata.qa.EditInspector;
+import org.openrefine.wikidata.qa.QAWarning;
+import org.openrefine.wikidata.qa.QAWarningStore;
+import org.openrefine.wikidata.schema.WikibaseSchema;
+import org.openrefine.wikidata.updates.ItemUpdate;
+import org.openrefine.wikidata.utils.FirstLinesExtractor;
+
+import com.google.refine.model.Project;
+import com.google.refine.util.ParsingUtilities;
+
+public class PreviewWikibaseSchemaCommand extends Command {
+
+    @Override
+    public void doPost(HttpServletRequest request, HttpServletResponse response)
+            throws ServletException, IOException {
+
+        try {
+            Project project = getProject(request);
+
+            response.setCharacterEncoding("UTF-8");
+            response.setHeader("Content-Type", "application/json");
+
+            String jsonString = request.getParameter("schema");
+
+            WikibaseSchema schema = null;
+            if (jsonString != null) {
+                try {
+                    JSONObject json = ParsingUtilities.evaluateJsonStringToObject(jsonString);
+                    schema = WikibaseSchema.reconstruct(json);
+                } catch (JSONException e) {
+                    respond(response, "error", "Wikibase schema could not be parsed.");
+                    return;
+                }
+            } else {
+                schema = (WikibaseSchema) project.overlayModels.get("wikibaseSchema");
+            }
+            if (schema == null) {
+                respond(response, "error", "No Wikibase schema provided.");
+                return;
+            }
+
+            QAWarningStore warningStore = new QAWarningStore();
+
+            // Evaluate project
+            Engine engine = getEngine(request, project);
+            List<ItemUpdate> editBatch = schema.evaluate(project, engine, warningStore);
+
+            StringWriter sb = new StringWriter(2048);
+            JSONWriter writer = new JSONWriter(sb);
+            writer.object();
+
+            {
```

Comment:
No need the {}. or  you want to extract into method?

Reply:
it's just meant to provide some structure to match the JSON's own structure (but I can remove it if you prefer, it is cosmetic)